src/StructureFactors/SampleGeneration.jl

```diff
@@ -113,12 +113,16 @@ function accum_sample!(sf::StructureFactor)
     nsamples[1] += 1
 
     # Transfer to final memory layout while accumulating new samplebuf
-    for ω in 1:nω, cell in CartesianIndices(latsize), j in 1:natoms, i in 1:natoms
+    cache = zeros(ComplexF64, size(data)[4:7])
+    for j in 1:natoms
+      for i in 1:natoms
         for (ci, c) in idxinfo 
-            α, β = ci.I
-            old = data[c,i,j,cell,ω]
-            data[c,i,j,cell,ω] = old + (samplebuf[α,cell,i,ω] * conj(samplebuf[β,cell,j,ω]) - old) / nsamples[1]
+          α, β = ci.I
+          @. cache = @views samplebuf[α,:,:,:,i,:] * conj(samplebuf[β,:,:,:,j,:]) - data[c,i,j,:,:,:,:] 
```

Comment:
Is it necessary to have a cache here? Could the division by `nsamples[1]` included in a one-liner to update `data`?

Reply:
Division is separated out here so there is one float divide and N float multiplys rather than N float divisions (minor speed boost). Could be `data[...] .+= copybuf .* (1/nsamples[1])` instead

Reply:
> Could be `data[...] .+= copybuf .* (1/nsamples[1])` instead

I like this version. Also worth checking whether the Julia compiler optimizes that down to `data[...] .+= copybuf ./ nsamples[1])`.

Reply:
I verified your result. The compiler will not perform the optimization, even with `@fastmath` enabled.

```julia

function f(a, b)
    @fastmath a .= a ./ b
end

function g(a, b)
    a .= a .* (1/b)
end

@code_native f(a, b)
@code_native g(a, b)
```


```